### ISSUE TITLE
Add muffling upgrade support to ALL machines and generators

### ISSUE THREAD

**Issue by Chaoschaot232** [mekanism/Mekanism-Feature-Requests#150]
**Describe the the feature you'd like**

Like the title states, some machines have sounds which makes me aggressive, like the Gas-Burning Generator or the Electrolytic Seperator. But since Mek has it's own sound system even in 1.12.2 Mufflers from ExU2 aren't working.

Well, for the ES I can use the muffling upgrade which is nice but the GBG doesn't even has the upgrade option on it's GUI so I
assume that this part of the mod isn't implemented to that machine (intended?).

**Describe alternatives you've considered**

Like stated abouve the hook up to the Upgrade API for Muffler Upgrades or any compatiblity to other mods Sound Mufflers.

**Additional context**

This looks for me like it was forgotten to be hooked into that muffling upgrade system even if used only in early game.
Tbh, I am on of the kind of players, which want to use Energy Sub Systems. This means I use besides high tier energy profuction also low tier energy profuction in a controlled environment.

In general it means that I have not only one energy production setup, I have different ones and all controlled over OpenComputers (if possible) or other controlling (e. g. ProjectRed).

All the generated Energy flows in different capacitors (any type of energy storage) and if the affected storages runs out of energy, the system will activate other energy producing systems starting from low tier to high tier ones. Well, the high tier ones will be directly used for e. g. maintaining my RFTools Dims (that energy hungry bastards ^^) and acting as the main fallback source if other low tier producers having trouble (they send out a special signal then).

Long Story short: I think you know what I mean and how I handle my energy system. So it would be nice to have this added to all machines, which don't have this upgrade supported yet and are not supporting any other mod's mufflers.

**Comment by Chaoschaot232** [comment]
That is nice to read.

Well, we are at Mekanism-1.12.2-9.8.3.390 so it should be the latest 1.12.2 but like stated ExU2 Mufflers are not working. I have read about another issiue here, that's because of the sound system why ExU2 Mufflers arn't working.

Will think about adding Super Sound Mufflers: Revived as the original one has gone on cures. Hopefully it isn't needed on the server side, as sounds are compleatly client side only, am I right?

**Comment by pupnewfster** [comment]
If memory serves it is needed server side mostly because it adds a block that is placed for muffling sounds.

**Comment by Chaoschaot232** [comment]
You are right, have looked into the source and found a few things wich are not rly relevant like the hard dependencies for The One Probe and Waila. After all, I doon't understand why it adds items/blocks if it seems to go over the specified class names to "muffle" them from what I have seen. Should be better then to make it client side only with a hotkey for it to open up the gui and setup what sounds you want to mute so that they are muted when you are arrounf them.

Will see what the current maintainer will responf to my request.
